### ISSUE TITLE
Bug: typescript compile errors when `exactOptionalPropertyTypes` compiler option is enabled

### ISSUE THREAD

**Issue by DetachHead** [typescript-eslint/typescript-eslint#6965]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

utils

### Repro Code

```typescript
import { TSESTree } from '@typescript-eslint/utils'
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "target": "es2022",
    "moduleResolution": "node",
    "strict": true,
    "exactOptionalPropertyTypes": true,
  }
}
```

### Expected Result

no compile errors

### Actual Result

```
node_modules/@typescript-eslint/types/dist/generated/ast-spec.d.ts:395:26 - error TS2430: Interface 'ClassExpression' incorrectly extends interface 'ClassBase'.
  Types of property 'abstract' are incompatible.
    Type 'undefined' is not assignable to type 'boolean'.

395 export declare interface ClassExpression extends ClassBase {
                             ~~~~~~~~~~~~~~~

node_modules/@typescript-eslint/types/dist/generated/ast-spec.d.ts:1510:19 - error TS2430: Interface 'TSModuleDeclarationModuleBase' incorrectly extends interface 'TSModuleDeclarationBase'.
  Types of property 'global' are incompatible.
    Type 'undefined' is not assignable to type 'boolean'.

1510 declare interface TSModuleDeclarationModuleBase extends TSModuleDeclarationBase {
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/@typescript-eslint/types/dist/generated/ast-spec.d.ts:1532:26 - error TS2430: Interface 'TSModuleDeclarationNamespace' incorrectly extends interface 'TSModuleDeclarationBase'.
  Types of property 'global' are incompatible.
    Type 'undefined' is not assignable to type 'boolean'.

1532 export declare interface TSModuleDeclarationNamespace extends TSModuleDeclarationBase {
                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 3 errors in the same file, starting at: node_modules/@typescript-eslint/types/dist/generated/ast-spec.d.ts:395
```

### Additional Info

enabling `skipLibCheck` will suppress the errors, but that's not ideal since it disables typechecking on all `.d.ts` files

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/utils`             | `5.59.1` |
| `TypeScript`                           | `4.9.5` |

**Comment by bradzacher** [comment]
duplicate of #6307. Please use the search.

**Comment by bradzacher** [comment]
The search term I used was `exactOptionalPropertyTypes`, which was the keyword you used in your title.

**Comment by DetachHead** [comment]
i probably did the search before i narrowed down exactly what compiler option was causing it.

maybe you guys should do what the typescript repo does and have a "search terms" section in the issue template, because that "I have searched for related issues" checkbox is useless if the maintainers don't trust the user when they check it:
```markdown
### 🔎 Search Terms

<!--
  What search terms did you use when trying to find an existing bug report?
  List them here so people in the future can find this one more easily.
-->
```
